src/apps/vpn/controller.cpp

```diff
@@ -576,7 +578,15 @@ void Controller::handshakeTimeout() {
 
 void Controller::disconnected() {
   logger.debug() << "Disconnected from state:" << m_state;
+  mozilla::glean::session::session_end.set();
 
+  // TODO: This must be after submission of ping - ensure it comes after the submissino
+
+  // We never expect to see this UUID in Glean metrics, as it is also
```

Comment:
One of the triggers is to send the ping right before the start of a new session. So this is false.

Reply:
Not according to the tech doc that this is based on: https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/184419319/Metric+pings+-+Tech+spec#Session-ping

If you have a problem with that doc, let's discuss it on the doc.

Reply:
Aha! Yes I have a problem with that indeed! Thanks for flagging it Matt, it was an oversight on my part not to comment on my first pass.

Reply:
Glad you noticed this! I've updated the comment here to reflect our updated plan for ping submission.